test/PositionManager.ts

```diff
@@ -123,18 +171,167 @@ describe("GammaPoolFactory", function () {
             }
 
             const res = await (await posMgr.withdrawNoPull(DepositWithdrawParams)).wait();
-            res.events.forEach(function(event: any, i: any){
-                if(i == 0)
-                    return;
-                console.log("event >>");
-                console.log(event.args);
-                expect(event.args.pool).to.equal(gammaPool.address);
-                expect(event.args.assets.toNumber()).to.equal(16);
-            });
-        });/**/
+            
+            const { args } = res.events[1];
+            expect(args.pool).to.equal(gammaPool.address);
+            expect(args.assets.toNumber()).to.equal(16);
+        });
+
+        it("#depositReserves should return shares and length of reserves", async function () {
+            await cfmm.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            const DepositReservesParams =  {
+                cfmm: cfmm.address,
+                amountsDesired: [10000, 100],
+                amountsMin: [1000, 10],
+                to: addr4.address,
+                protocol: 1,
+                deadline: ethers.constants.MaxUint256
+            }
+            const res = await (await posMgr.depositReserves(DepositReservesParams)).wait();
+            
+            const { args } = res.events[0];
+            expect(args.pool).to.equal(gammaPool.address);
+            expect(args.reservesLen).to.equal(4);
+            expect(args.shares.toNumber()).to.equal(18);
+        });
+
+        it("#withdrawReserves should return assets and lenght of reserves", async function () {            
+            const WithdrawReservesParams =  {
+                cfmm: cfmm.address,
+                protocol: 1,
+                amount: 1000,
+                amountsMin: [100, 200, 300],
+                to: owner.address,
+                deadline: ethers.constants.MaxUint256
+            }
+
+            const res = await (await posMgr.withdrawReserves(WithdrawReservesParams)).wait();
+
+            const { args } = res.events[1];
+            expect(args.pool).to.equal(gammaPool.address);
+            expect(args.reservesLen).to.equal(3);
+            expect(args.assets.toNumber()).to.equal(17);
+        });
     });
+
     // You can nest describe calls to create subsections.
-    describe("Long Gamma", function () {
+    describe("Long Gamma Functions", function () {
+        it("#createLoan should return tokenId", async function () {
+            expect(tokenId.toNumber()).to.equal(1);
```

Comment:
You must call posMgr.createLoan here. Don't use tokenId. just make sure createLoan returns a number different than 1 or it will create issues.